packages/core/src/entrypoint/0.6.ts

```diff
@@ -1,28 +1,115 @@
-import type { Address, Chain } from "viem";
-import type { UserOperationRequest } from "../types";
 import {
-  getDefaultEntryPointAddress,
-  getUserOperationHash,
-  packUo,
-} from "../utils/index.js";
-import type { EntryPointDef } from "./types";
-
-export const getVersion060EntryPoint: (
-  chain: Chain,
-  address?: Address
-) => EntryPointDef<UserOperationRequest> = (
-  chain: Chain,
-  address = getDefaultEntryPointAddress(chain)
-) => {
-  return {
-    version: "0.6.0",
-    address,
-    chain,
-    packUserOperation: (userOperation: UserOperationRequest) => {
-      return packUo(userOperation);
-    },
-    getUserOperationHash: (userOperation: UserOperationRequest) => {
-      return getUserOperationHash(userOperation, address, chain.id);
-    },
-  } satisfies EntryPointDef<UserOperationRequest>;
+  encodeAbiParameters,
+  hexToBigInt,
+  keccak256,
+  type Address,
+  type Chain,
+  type Hash,
+  type Hex,
+} from "viem";
+import {
+  arbitrum,
+  arbitrumGoerli,
+  arbitrumSepolia,
+  base,
+  baseGoerli,
+  baseSepolia,
+  goerli,
+  mainnet,
+  optimism,
+  optimismGoerli,
+  optimismSepolia,
+  polygon,
+  polygonAmoy,
+  polygonMumbai,
+  sepolia,
+} from "viem/chains";
+import { EntryPointAbi_v6 } from "../abis/EntryPointAbi_v6.js";
+import type {
+  UserOperationLike,
+  UserOperationRequest,
+  UserOperationStruct,
+} from "../types.js";
+import type { SupportedEntryPoint } from "./types.js";
+
+const packUserOperation = (request: UserOperationRequest<"0.6.0">): Hex => {
+  const hashedInitCode = keccak256(request.initCode);
+  const hashedCallData = keccak256(request.callData);
+  const hashedPaymasterAndData = keccak256(request.paymasterAndData);
+
+  return encodeAbiParameters(
+    [
+      { type: "address" },
+      { type: "uint256" },
+      { type: "bytes32" },
+      { type: "bytes32" },
+      { type: "uint256" },
+      { type: "uint256" },
+      { type: "uint256" },
+      { type: "uint256" },
+      { type: "uint256" },
+      { type: "bytes32" },
+    ],
+    [
+      request.sender as Address,
+      hexToBigInt(request.nonce),
+      hashedInitCode,
+      hashedCallData,
+      hexToBigInt(request.callGasLimit),
+      hexToBigInt(request.verificationGasLimit),
+      hexToBigInt(request.preVerificationGas),
+      hexToBigInt(request.maxFeePerGas),
+      hexToBigInt(request.maxPriorityFeePerGas),
+      hashedPaymasterAndData,
+    ]
+  );
 };
+
+export default {
+  version: "0.6.0",
+
+  address: {
```

Comment:
I really think we should avoid any hardcoding of networks in "core." Is there a way around this?

Reply:
We could just assume the same address for all networks. The things to keep an eye on:

* If some network uses a different address. I don't know why that would happen, but it's possible.
* We'd need handling for when a network doesn't have an entry point deployed. Potentially we'd have a check somewhere that this address has code on the selected network.

Definitely agree that not hardcoding networks would be great. I think if we didn't need this mapping, we could simplify some types by not needing a `TChain` type parameter on `SupportedEntryPoint` and related types.

Reply:
I'm ok with hardcoding some defaults, as long as we have some means of overwriting them like we do for account addresses, factories, etc

Reply:
@denniswon how do we handle chain and address overrides here? what happens if we don't support a chain in the SDK yet, but someone wants to use the EP on a chain our infra or their infra supports?

Reply:
bump